lib/src/lib/dataset.ts

```diff
@@ -0,0 +1,170 @@
+        }
+      } else {
+        const nonIgnoredFiles = (await this.fromDirectory(tmpDir)).ok();
+        const ignoredFile = undefined;
+        return Result.ok({ nonIgnoredFiles, ignoredFile });
+      }
+    } finally {
+      await fs.rm(tmpDir, { recursive: true });
+    }
+  }
+
+
+  private async fromCSV(
+    infoPath: string,
+    ignore?: string
+  ): Promise<Result<{ nonIgnoredFiles: File[], ignoredFile: File | undefined }>> {
+    const dirname = path.dirname(infoPath);
+    try {
+      const files = csvParse((await fs.readFile(infoPath)).toString())
+        .map((row:  DSVRowString) => ({
+          filename: row.filename as string,
+          fullName: row.full_name as string,
+          id: row.id as string,
+          status: row.status as string,
+          submissionID: row.submission_id as string,
+          nameEN: row.name_en as string,
+          nameNL: row.name_nl as string,
+          exerciseID: row.exercise_id as string,
+          createdAt: new Date(row.created_at as string),
+          labels: row.label as string || row.labels as string,
+          ignored: row.ignored as string
+        }))
+        .map((row: ExtraInfo) => readPath(path.join(dirname, row.filename), row));
+      const resolvedFiles = await Result.all(files);
+      const ignoredFile = await this.setIgnoredFile(resolvedFiles.ok(), ignore);
+      const nonIgnoredFiles = resolvedFiles.ok().filter(file => file.extra?.ignored !== "true");
+      return Result.ok({ nonIgnoredFiles, ignoredFile });
+    } catch(e) {
+      throw new Error("The given '.csv'-file could not be opened");
+    }
+  }
+
+
+  public async createNameCandidate(paths: string[]): Promise<string> {
+    let nameCandidate = "undefined";
+    if (paths.length === 1) {
+      if (paths[0].toLowerCase().endsWith(".zip")) {
+        nameCandidate = path.basename(paths[0], ".zip");
+      } else if (paths[0].toLowerCase().endsWith(".csv")) {
+        if (paths[0].endsWith("info.csv")) {
+          nameCandidate = path.dirname(paths[0]).split(path.sep).pop() || "undefined";
+        }
+      }
+    }
+    else if (paths.length === 2) {
+      nameCandidate = path.basename(paths[0]) + " & " + path.basename(paths[1]);
+    }
+    return nameCandidate;
+  }
+
+  public async prepareDataset(paths: string[], ignore?: string): Promise<Dataset> {
+    let files = null;
+    let filesFromZIP = null;
+    let filesFromCSV = null;
+  
+    if (paths.length == 1) {
+      const inputFile = paths[0];
+      if (inputFile.toLowerCase().endsWith(".zip")) {
+        filesFromZIP = this.fromZIP(inputFile, ignore);
+      } else if (inputFile.toLowerCase().endsWith(".csv")) {
+        filesFromCSV = this.fromCSV(inputFile, ignore);
+      } else {
+        throw new Error("You gave one input file, but it is not a CSV file or a ZIP archive.");
+      }
+    } else {
+      files = readFiles(paths);
+    }
+
+    this.nameCandidate = await this.createNameCandidate(paths);
+    
+    if (files) {
```

Comment:
The construct of checking which situation we have (ZIP-file, CSV-file or paths) occurs in multiple locations now. Maybe we can group those together in the single function for each situation?

For example:

```ts
if (inputFile.toLowerCase().endsWith(".zip") {
	return Dataset.fromZIP(inputFile, ignored);
}
```

and then the `fromZIP` function would be another static function that would create the `Dataset` instance with the `name`, `files`, etc. set.